homeassistant/components/dsmr/config_flow.py

```diff
@@ -223,7 +222,7 @@ async def async_step_setup_serial(
                 return await self.async_step_setup_serial_manual_path()
 
             dev_path = await self.hass.async_add_executor_job(
-                get_serial_by_id, user_selection
+                usb.get_serial_by_id, user_selection
```

Comment:
It might make sense to hide the executor call behind a `usb.async_get_serial_by_id `